javascript/javascript3/week1/Giphy-Api/script.js

```diff
@@ -0,0 +1,55 @@
+let searchKey = document.getElementById('getGif')
+let getGifBtn = document.getElementById('btnSearch')
+let getNumber = document.getElementById('getNumber')
+let pictureBox = document.getElementById('pictureBox')
+
+getGifBtn.addEventListener('click', giphy)
+
+getNumber.addEventListener('input', giphy1)
+
+async function giphy() {
+
+    pictureBox.innerHTML = '';
+
+    let search = searchKey.value;
+
+    let image = await fetch(`https://api.giphy.com/v1/gifs/search?api_key=92CWtdMWMOYaLWV6sKMC3WZgM6grBT4l&q=${search}&limit=50&offset=0&rating=g&lang=en`)
```

Comment:
instead of writing this part twice, you can have a function to fetch data for you like:
const giphyData = async (searchKey, searchNumber) => {
  const API_KEY = "92CWtdMWMOYaLWV6sKMC3WZgM6grBT4l";
  const data = await fetch(
    `https://api.giphy.com/v1/gifs/search?api_key=${API_KEY}&q=${searchKey}&limit=${searchNumber}&offset=0&rating=g&lang=en`
  ).then((response) => response.json());
  return data;
};

Reply:
I have a problem with this way. When I use function in the same way yo mentioned, It will give me a promise instead of object. Whould you please help me. I have already changed first function as you mentioned here and commit it.

Reply:
I got it. Thank you so much...